### ISSUE TITLE
Adding Systemd support for transmission

### ISSUE THREAD

**Issue by bockbilbo** [karamanolev/WhatManager2#60]
Hey! 

I've notice that the new transmission instance script creates upstart and not systemd scripts. Ubuntu, Debian, CentOS and many other distros have migrated to systemd, so I think it might be worth adding support to systemd.

I have tried the following with my instances (and it works):

For each instance, create a file like _/etc/systemd/system/transmission-passtheheadphones01.service_ with the following content:
 
```
[Unit]
Description=PTH Transmission Daemon 1
After=syslog.target network.target
 
[Service]
Type=simple
User=transmission-passtheheadphones01
StandardError=syslog
ExecStart=/usr/local/bin/transmission-daemon -f -g /opt/wm2/data/transmission/passtheheadphones01
Restart = always
RestartSec = 240
 
[Install]
WantedBy=multi-user.target
```
 
Then:

- Update permissions:
  chmod 664 /etc/systemd/system/transmission-passtheheadphones01.service
- Reload systemctl daemon so that it knows about the file:
  sudo systemctl daemon-reload
- Add the boot flag to the service:
  sudo systemctl enable transmission-passtheheadphones01.service
- Start the service:
  sudo systemctl start transmission-passtheheadphones01.service

I hope this is enough info for you guys to update the script if you wish to.

Thanks a lot!

**Comment by bockbilbo** [comment]
This will also need to set the following settings:

sudo sysctl -w net.core.rmem_max=4194304
sudo sysctl -w net.core.wmem_max=1048576

**Comment by bockbilbo** [comment]
Please, disregard everything explained above. The settings did not work properly after reboot and did not have time to investigate. It turned out that Ubuntu 16.04, after running the transmission_new script, requires running _sudo systemctl daemon-reload_ to be able to start the process.

**Comment by bockbilbo** [comment]
It might be a good idea to update the transmisison_new script to run "sudo systemctl daemon-reload" if systemctl is in the user´s path.

**Comment by casedefault** [comment]
https://github.com/karamanolev/WhatManager2/pull/72 should sort this out.

**Comment by karamanolev** [comment]
@bockbilbo If you can confirm that this is working, I'd be OK merging it. Are you still using WM2?

**Comment by bockbilbo** [comment]
Hi @karamanolev I still use wm2 and use systemd for the transmission instances.

I have made some changes since I posted my original ticket:

1. I use separate transmission configuration folders for each instance at /var/lib/transmission-daemon-redXX - replacing XX with the instance number and making sure the folder is owned by the debian-transmission user (in Ubuntu 20.04)
2. I create separate systemd files named /etc/systemd/system/transmission-daemon-redXX.service with the following content:

```
[Unit]
Description=Transmission BitTorrent Daemon #XX for Redacted
After=network.target

[Service]
User=debian-transmission
Type=notify
ExecStart=/usr/bin/transmission-daemon -f --log-error -g /var/lib/transmission-daemon-redXX
ExecStop=/bin/kill -s STOP $MAINPID
ExecReload=/bin/kill -s HUP $MAINPID

[Install]
WantedBy=multi-user.target
```

Not sure though that @casedefault's solution is the same.

Thank you!
